drivers/serial/eusart_gecko.c

```diff
@@ -0,0 +1,532 @@
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	.fifo_fill = eusart_gecko_fifo_fill,
+	.fifo_read = eusart_gecko_fifo_read,
+	.irq_tx_enable = eusart_gecko_irq_tx_enable,
+	.irq_tx_disable = eusart_gecko_irq_tx_disable,
+	.irq_tx_complete = eusart_gecko_irq_tx_complete,
+	.irq_tx_ready = eusart_gecko_irq_tx_ready,
+	.irq_rx_enable = eusart_gecko_irq_rx_enable,
+	.irq_rx_disable = eusart_gecko_irq_rx_disable,
+	.irq_rx_ready = eusart_gecko_irq_rx_ready,
+	.irq_err_enable = eusart_gecko_irq_err_enable,
+	.irq_err_disable = eusart_gecko_irq_err_disable,
+	.irq_is_pending = eusart_gecko_irq_is_pending,
+	.irq_update = eusart_gecko_irq_update,
+	.irq_callback_set = eusart_gecko_irq_callback_set,
+#endif
+};
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+#define GECKO_EUSART_IRQ_HANDLER_DECL(idx)				       \
+	static void eusart_gecko_config_func_##idx(const struct device *dev)
+#define GECKO_EUSART_IRQ_HANDLER_FUNC(idx)				       \
+	.irq_config_func = eusart_gecko_config_func_##idx,
+#define GECKO_EUSART_IRQ_HANDLER(idx)					       \
+	static void eusart_gecko_config_func_##idx(const struct device *dev)   \
+	{								       \
+		IRQ_CONNECT(DT_INST_IRQ_BY_NAME(idx, rx, irq),		       \
+			    DT_INST_IRQ_BY_NAME(idx, rx, priority),	       \
+			    eusart_gecko_isr, DEVICE_DT_INST_GET(idx), 0);     \
+		IRQ_CONNECT(DT_INST_IRQ_BY_NAME(idx, tx, irq),		       \
+			    DT_INST_IRQ_BY_NAME(idx, tx, priority),	       \
+			    eusart_gecko_isr, DEVICE_DT_INST_GET(idx), 0);     \
+									       \
+		irq_enable(DT_INST_IRQ_BY_NAME(idx, rx, irq));		       \
+		irq_enable(DT_INST_IRQ_BY_NAME(idx, tx, irq));		       \
+	}
+#else
+#define GECKO_EUSART_IRQ_HANDLER_DECL(idx)
+#define GECKO_EUSART_IRQ_HANDLER_FUNC(idx)
+#define GECKO_EUSART_IRQ_HANDLER(idx)
+#endif
+
+#define GECKO_EUSART_INIT(idx)							\
+GECKO_EUSART_IRQ_HANDLER_DECL(idx);						\
+										\
+PINCTRL_DT_INST_DEFINE(idx);							\
+										\
+static struct uart_config uart_cfg_##idx = {					\
+	.baudrate  = DT_INST_PROP_OR(idx, current_speed,			\
+				     GECKO_UART_DEFAULT_BAUDRATE),		\
+	.parity    = DT_INST_ENUM_IDX_OR(idx, parity,				\
+					 GECKO_UART_DEFAULT_PARITY),		\
+	.stop_bits = DT_INST_ENUM_IDX_OR(idx, stop_bits,			\
+					 GECKO_UART_DEFAULT_STOP_BITS),		\
+	.data_bits = DT_INST_ENUM_IDX_OR(idx, data_bits,			\
+					 GECKO_UART_DEFAULT_DATA_BITS),		\
+	.flow_ctrl = DT_INST_PROP(idx, hw_flow_control)				\
+					? UART_CFG_FLOW_CTRL_RTS_CTS		\
+					: UART_CFG_FLOW_CTRL_NONE,		\
+};										\
+										\
+static const struct eusart_gecko_config eusart_gecko_cfg_##idx = {		\
+	.eusart = (EUSART_TypeDef *)DT_INST_REG_ADDR(idx),			\
+	.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(idx),				\
+	.clock_dev = DEVICE_DT_GET(DT_INST_CLOCKS_CTLR(idx)),			\
+	.clock_cfg = SILABS_DT_INST_CLOCK_CFG(idx),				\
+	GECKO_EUSART_IRQ_HANDLER_FUNC(idx)					\
+	};									\
+										\
+static struct eusart_gecko_data eusart_gecko_data_##idx = {			\
+	.uart_cfg = &uart_cfg_##idx,						\
```

Comment:
Why do you need for `uart_cfg_##idx`? You can directly initialize `uart_cfg`.

Reply:
This is how it's done in other drivers

Reply:
done